File: go/pkg/model/priority.go
package model

// Priority represents task priority levels
// References: backend/models.py Priority enum
//            frontend/src/types.ts Priority enum
//            java/src/main/java/com/taskmanager/model/Priority.java
type Priority string

const (
	PriorityLow    Priority = "low"
	PriorityMedium Priority = "medium"
	PriorityHigh   Priority = "high"
)

// String returns the string representation of Priority
func (p Priority) String() string {
	return string(p)
}

// FromString creates a Priority from a string value
func PriorityFromString(s string) Priority {
	switch s {
	case "low":
		return PriorityLow
	case "high":
		return PriorityHigh
	default:
		return PriorityMedium
	}
}

File: java/src/main/java/com/taskmanager/model/Priority.java
package com.taskmanager.model;

/**
 * Priority enum
 * References: backend/models.py (Python Priority enum)
 *            frontend/src/types.ts (TypeScript Priority enum)
 */
public enum Priority {
    LOW("low"),
    MEDIUM("medium"),
    HIGH("high");

    private final String value;

    Priority(String value) {
        this.value = value;
    }

    public String getValue() {
        return value;
    }

    public static Priority fromString(String value) {
        for (Priority priority : Priority.values()) {
            if (priority.value.equalsIgnoreCase(value)) {
                return priority;
            }
        }
        return MEDIUM; // Default
    }
}

